backend/models/class.model.ts

```diff
@@ -81,4 +82,20 @@ const ClassSchema: Schema = new Schema(
   { timestamps: true },
 );
 
+/* eslint-disable func-names */
+ClassSchema.pre("findOneAndDelete", async function (next) {
+  try {
+    const doc = await this.findOne(this.getQuery()).clone();
+    if (doc) {
+      // Delete all test sessions associated with class
+      /* eslint-disable no-underscore-dangle */
+      await MgTestSession.deleteMany({ class: doc._id });
```

Comment:
Does `id` not exist on the object at this point? Not sure how mongoose treats objects at this point.